### ISSUE TITLE
extraconfig parameter unable to be set via Domain Admin

### ISSUE THREAD

**Issue by scottsignal** [apache/cloudstack#9397]
##### ISSUE TYPE

 * Improvement Request

##### COMPONENT NAME

~~~
API
~~~

##### CLOUDSTACK VERSION

~~~
4.19.02
~~~

##### CONFIGURATION

Ubuntu 22.04
KVM

##### OS / ENVIRONMENT

N/A

##### SUMMARY

A user in an account with the default out of the box role "Domain Admin" is unable to add the extraconfig parameter via API.  Running this same API command as Root Admin works just fine.

##### STEPS TO REPRODUCE

Ensure that the 2 options are set and reboot mgmt:
allow.additional.vm.configuration.list.kvm = backend, tpm, devices
Enable additional vm configuration = enabled
Build an account & user with the out of the box role "Domain Admin"
Build an instance normally, however, make sure the Start Instance ratio is unselected at the bottom.
Run the API command to modify add the extra config.  Running this command does not work under an account with out of the box default "Domain Admin" role.  

~~~
https://example.com/client/api?command=updateVirtualMachine&id=<ID_Here>&extraconfig=%3Cdevices%3E%0A%3Ctpm%20model%3D%22tpm-tis%22%3E%0A%3Cbackend%20type%3D%22emulator%22%20version%3D%222.0%22%2F%3E%0A%3C%2Ftpm%3E%0A%3C%2Fdevices%3E
~~~

##### EXPECTED RESULTS

extraconfig added to the VM.

##### ACTUAL RESULTS

API returns results with no error, however, extraconfig is not added.

**Comment by weizhouapache** [comment]
@scottsignal 
it is expected, only root admin can set extraconfig

refer to 

https://github.com/apache/cloudstack/blob/afdf4d7d46c0ce4e85da12140f9e140d5f4fe490/api/src/main/java/org/apache/cloudstack/api/command/user/vm/UpdateVMCmd.java#L144-L145

it has `authorized = { RoleType.Admin }`

**Comment by weizhouapache** [comment]
we could remove the restriction, as the extraconfig is allowed when deploy a vm

**Comment by scottsignal** [comment]
> we could remove the restriction, as the extraconfig is allowed when deploy a vm

Unless there is good reason to not.  At a minimum, it would be nice to allow that not to be hard coded and be able to select roles to be selected in global settings for additional roles to add additional extraconfig.  Thanks!

**Comment by nvazquez** [comment]
I agree with @scottsignal - also there are global settings which can restrict the passing of the extra configs to the hypervisor side even though the extra configs are defined as VM settings:

![image](https://github.com/user-attachments/assets/a9df14ca-748e-4af3-96a5-6d256e76cb9c)

**Comment by DaanHoogland** [comment]
Ok, so we can change the hard-coding to allow access to the parameter for all and "just" check the global settings during the call, right @nvazquez ?

**Comment by nvazquez** [comment]
Yes @DaanHoogland - the access is already in place using these configurations as far as I know so only editing the access should work

**Comment by DaanHoogland** [comment]
fixed in #9422
